api/tests/unit/integrations/github/test_unit_github_views.py

```diff
@@ -776,3 +780,209 @@ def test_cannot_fetch_repositories_when_there_is_no_installation_id(
+
+    # Add response for endpoint being tested
+    responses.add(
+        method=responses.GET,
+        url=(
+            f"{GITHUB_API_URL}repos/{github_repository.repository_owner}/{github_repository.repository_name}/"
+            "contributors?&per_page=100&page=1"
+        ),
+        json=mocked_github_response,
+        status=200,
+    )
+
+    # When
+    response = admin_client_new.get(
+        path=url,
+        data={
+            "repo_owner": github_repository.repository_owner,
+            "repo_name": github_repository.repository_name,
+        },
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json() == expected_response
+
+
+def test_fetch_github_repo_contributors_with_invalid_query_params(
+    admin_client_new: APIClient,
+    organisation: Organisation,
+    github_configuration: GithubConfiguration,
+    github_repository: GithubRepository,
+) -> None:
+    # Given
+    url = reverse(
+        viewname="api-v1:organisations:get-github-repo-contributors",
+        args=[organisation.id],
+    )
+
+    # When
+    response = admin_client_new.get(
+        path=url,
+        data={
+            "repo_owner": github_repository.repository_owner,
+        },
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.json() == {"error": {"repo_name": ["This field is required."]}}
+
+
+def test_github_api_call_error_handler_with_value_error(
+    mocker: MockerFixture,
+) -> None:
+    # Given
+    @github_api_call_error_handler()
+    def test_view(request):
+        raise ValueError("Invalid parameter")
+
+    # When
+    response = test_view(None)
+
+    # Then
+    assert isinstance(response, Response)
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert response.data == {
+        "detail": "Failed to retrieve requested information from GitHub API. Error: Invalid parameter"
+    }
+
+
+@pytest.mark.parametrize(
+    "page, page_size, is_invalid_page",
+    [
+        (
+            1,
+            103,
+            False,
+        ),
+        (
+            0,
+            100,
+            True,
+        ),
+    ],
+)
+def test_send_the_invalid_number_page_or_page_size_param_returns_400(
+    admin_client: APIClient,
+    organisation: Organisation,
+    github_configuration: GithubConfiguration,
+    github_repository: GithubRepository,
+    page: int,
+    page_size: int,
+    is_invalid_page: bool,
+) -> None:
+    # Given
+    data: dict[str, str | int] = {
+        "installation_id": github_configuration.installation_id,
+        "page": page,
+        "page_size": page_size,
+    }
+
+    url = reverse(
+        "api-v1:organisations:get-github-installation-repos", args=[organisation.id]
+    )
+    # When
+    response = admin_client.get(url, data)
+
+    # Then
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    response_json = response.json()
+    if is_invalid_page:
```

Comment:
An `if` inside a test is a code smell. I think it's better to have a parametrised `detail` or `response_json` argument instead.